tests/test_esm_runscripts/test_filedicts.py

```diff
@@ -633,3 +633,207 @@ def test_resolve_paths_old_config():
     assert sim_file["absolute_path_in_run_tree"] == Path(
         "/work/ollie/mandresm/testing/run/awicm3//awicm3-v3.1-TCO95L91-CORE2_initial/run_20000101-20000101/input/oifs/o3chem_l91"
     )
+
+def test_wild_card_check():
+    """Tests that wild card check is passed"""
+    source_name = "a_wild_card*name*.txt"
+    target_name = "another_wild_card*newname*.txt1"
+    source_pattern = source_name.split("*")
+    target_pattern = target_name.split("*")
+
+    assert(esm_runscripts.filedicts.SimulationFile.wild_card_check(
+        source_pattern, target_pattern
+    ))
+
+def test_wild_card_check_fails():
+    """
+    Tests that, when given an incorrect wildcard pattern, a ``user_error`` is reported
+    """
+    source_name = "a_wild_card*name*.txt"
+    target_name = "another_wild_cardnewname*.txt1"
+    source_pattern = source_name.split("*")
+    target_pattern = target_name.split("*")
+
+    # Captures output (i.e. the user-friendly error)
+    with Capturing() as output:
+        with pytest.raises(SystemExit) as error:
+            esm_runscripts.filedicts.SimulationFile.wild_card_check(
+                    source_pattern, target_pattern
+            )
+
+    # error needs to occur as the path is not absolute
+    assert any(["The wild card pattern of the source" in line for line in output])
+
+def test_find_globbing_files(fs):
+    """
+    Tests for files with the globbing pattern not found
+    """
+
+    dummy_config = """
+    general:
+        thisrun_work_dir: /work/ollie/mandresm/awiesm/run_20010101-20010101/work/
+        all_model_filetypes: [analysis, bin, config, forcing, input, couple, log, mon, outdata, restart, viz, ignore]
+    oifs:
+        files:
+            oifsnc:
+                name_in_work: input_expid_*_DATE_*.nc
+                name_in_exp_tree: new_input_expid_*_NEW_DATE_*.nc
+                type: outdata
+        experiment_outdata_dir: /work/ollie/pgierz/some_exp/input/oifs
+        thisrun_outdata_dir: /work/ollie/pgierz/some_exp/run_20010101-20010101/input/oifs
+    """
+    config = yaml.safe_load(dummy_config)
+    files = [
+        "input_11_DATE_12.nc",
+        "input_21_DATE_22.nc",
+        "input_31_DATE_32.nc",
+    ]
```

Comment:
```suggestion
    config = yaml.safe_load(dummy_config)
    # These files are incorrect since they do not have input_expid and instead have input_<NUMBER>
    files = [
        "input_11_DATE_12.nc",
        "input_21_DATE_22.nc",
        "input_31_DATE_32.nc",
    ]
```